test/kafka/test/SeededBroker.scala

```diff
@@ -171,27 +171,23 @@ case class SimpleProducer(topic: String,
 
   props.put("compression.codec", codec.toString)
   props.put("producer.type", if(synchronously) "sync" else "async")
-  props.put("metadata.broker.list", brokerList)
+  props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+  props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
+  props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
+
   props.put("batch.num.messages", batchSize.toString)
   props.put("message.send.max.retries", messageSendMaxRetries.toString)
   props.put("request.required.acks",requestRequiredAcks.toString)
   props.put("client.id",clientId.toString)
 
-  val producer = new Producer[AnyRef, AnyRef](new ProducerConfig(props))
-
-  def kafkaMesssage(message: Array[Byte], partition: Array[Byte]): KeyedMessage[AnyRef, AnyRef] = {
-    if (partition == null) {
-      new KeyedMessage(topic,message)
-    } else {
-      new KeyedMessage(topic,partition,message)
-    }
-  }
+  val producer = new KafkaProducer[AnyRef, AnyRef](props)
 
-  def send(message: String, partition: String = null): Unit = send(message.getBytes("UTF8"), if (partition == null) null else partition.getBytes("UTF8"))
```

Comment:
again this `else partition.getBytes("UTF8")` removal looks like it would break backwards compatibility